src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2306,6 +2308,77 @@ namespace Idint {
 
 } 
 
+namespace Ishftc {
+
+    static ASR::expr_t *eval_Ishftc(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*> &args) {
+        int64_t val = ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_n;
+        int64_t shift = ASR::down_cast<ASR::IntegerConstant_t>(args[1])->m_n;
+        int kind = ASRUtils::extract_kind_from_ttype_t(ASR::down_cast<ASR::IntegerConstant_t>(args[0])->m_type);
+        int bits_size = 8*kind;
+        if (bits_size < shift) {
```

Comment:
`shift` here can be negative. So we should take absolute value.
```suggestion
        if (bits_size < abs(shift)) {
```

Also, I think the condition is more readable when it is written as `if (abs(shift) > bits_size)`